src/assets/src/hooks/useDialogState.ts

```diff
@@ -0,0 +1,30 @@
+import { useState } from "react";
+
+import { DialogState } from "../components/common";
+
+export function useDialogState (): [
+    DialogState, (title: string, description: string, action: () => void) => void
+] {
+    const [dialogState, setDialogState] = useState<DialogState>({ show: false });
+
+    const resetDialogState = () => {
+        setDialogState({ show: false })
```

Comment:
this take and Object of type `DialogState` and only thing that is updated is the action state. is it not needed to update the rest of the attribute like title etc? is this more like an append of the existing `DialogState` Object

Reply:
No, this clears all properties except for `show` and sets it to `false`. That sets it up for it to be used again, either by the same action or another one.